arch/arm/soc/nordic_nrf5/nrf52/soc_power.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _SOC_POWER_H_
+#define _SOC_POWER_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum power_states {
+	SYS_POWER_STATE_CPU_LPS,       /* CONSTLAT Task Activited */
+	SYS_POWER_STATE_CPU_LPS_1,     /* LOWPWR Task Activated */
+	SYS_POWER_STATE_CPU_LPS_2,     /* Not supported*/
+	SYS_POWER_STATE_DEEP_SLEEP,    /* System Off */
+	SYS_POWER_STATE_DEEP_SLEEP_1,  /* Not Supported */
+	SYS_POWER_STATE_DEEP_SLEEP_2,  /* Not Supported */
```

Comment:
Perhaps I'm missing some context here. What is the purpose of defining states that are not (and will not be) supported? Is the intention to make this enumerated type common for all the chips? But why is it then defined separately in the same form but in several places?

Reply:
I have keep consistency across the Chips.

Reply:
We are planning to have a "power.h" file which is same across architectures/Chips. So this duplication  of these states for different SOCs will be gone. This is for stop gap arrangement.